src/vs/workbench/api/browser/mainThreadTerminalService.ts

```diff
@@ -276,9 +276,9 @@ export class MainThreadTerminalService implements MainThreadTerminalServiceShape
 	}
 
 	public $sendProcessTitle(terminalId: number, title: string): void {
-		const terminalProcess = this._terminalProcessProxies.get(terminalId);
-		if (terminalProcess) {
-			terminalProcess.emitTitle(title);
+		const instance = this._terminalService.getInstanceFromId(terminalId);
+		if (instance) {
+			instance.setTitle(title, TitleEventSource.Api);
```

Comment:
It's a little confusing here how the _process_ title event gets routed as an API event, I get why it's needed though (so it replaces `name` and doesn't need `experimentalUseTitleEvent` set). I'll add a comment here